### DISCUSSION TITLE
Making the bot able to open and close channels

### DISCUSSION THREAD

**Discussion by Azzaare** [Humans-of-Julia/HoJBot.jl#38] [Ideas]
I would like to have the bot able to create and close temporary channels on a per-user basis.
For instance, assume one user wants to play a hojbot game for a while on day d.
He could ask for a personal game "channel" to be opened for some time (let's say an hour).

We could make the channel permanent, but I am not sure the channel can be hidden from the mods, which would make it pretty noisy.

Anyway, it could be useful for other circumstances, what do you think?

**Comment by tk3369** [comment]
My only concern is the potential of being abused, or to a lesser extent people forgetting to delete the temp channel. 

Perhaps it would be best to integrate with the game foe whatever it is needed? Then you can manage the lifecycle and destroy it automatically when it's no longer needed.

**Reply by Azzaare** [reply]
I was thinking that the bot would create the channel for short duration (maybe 10 minutes or so) and delete it.

What I was concerned about is privacy of information (for the game, but also for other things). I thought it could work a bit like the ticket notification bot Aru showed us some day. Except that the bot would close it after some time. The channel would only be accessible by the player requesting to "play" a game session and probably the mods.

How does it sound?

I am just looking for a way to interact privately with the bot for games, challenges, etc

Also, I plan to add a submission/PR feature for challenges with automated CI (so tested in a container) for the bot. The bot would be the one posting the PR on a private repo. etc.

**Comment by rmsrosa** [comment]
I don't like the idea of the bot spawning channels like that.

What if we create a specific channel for the bot, but as a private channel, and give the bot the ability to include and exclude users based on whether they want to start or stop playing the game? That would be cleaner.

👀 I thought that this was in the "Discussions". Anyways.

**Reply by Azzaare** [reply]
What do you mean? It was always in Discussions (obviously a lie ^^)
